test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -191,304 +94,330 @@ def setUp(self):
-                self.op_overload, "aten::add", self.diagnostic_context
+            (
+                function_overloads,
+                custom_overloads,
+            ) = self.dispatcher.get_function_overloads(
+                op_overload, "aten", "add", None, self.diagnostic_context
             )
             self.dispatcher._find_the_perfect_or_nearest_match_onnxfunction(
-                self.op_overload,
-                "aten::add",
+                op_overload,
+                "aten",
+                "add",
+                None,
                 function_overloads,
-                self.op_overload.args,
-                self.op_overload.kwargs,
+                custom_overloads,  # custom function overloads
+                op_overload.args,
+                op_overload.kwargs,
                 self.diagnostic_context,
             )
 
-    @common_utils.parametrize(
-        "available_opsets, target, expected",
+    @parameterized.expand(
         [
-            ((7, 8, 9, 10, 11), 16, 11),
-            ((7, 8, 9, 10, 11), 11, 11),
-            ((7, 8, 9, 10, 11), 10, 10),
-            ((7, 8, 9, 10, 11), 9, 9),
-            ((7, 8, 9, 10, 11), 8, 8),
-            ((7, 8, 9, 10, 11), 7, 7),
-            ((9, 10, 16), 16, 16),
-            ((9, 10, 16), 15, 10),
-            ((9, 10, 16), 10, 10),
-            ((9, 10, 16), 9, 9),
-            ((9, 10, 16), 8, 9),
-            ((9, 10, 16), 7, 9),
-            ((7, 9, 10, 16), 16, 16),
-            ((7, 9, 10, 16), 10, 10),
-            ((7, 9, 10, 16), 9, 9),
-            ((7, 9, 10, 16), 8, 9),
-            ((7, 9, 10, 16), 7, 7),
-            ([17], 16, None),  # New op added in 17
-            ([9], 9, 9),
-            ([9], 8, 9),
-            ([], 16, None),
-            ([], 9, None),
-            ([], 8, None),
-            # Ops registered at opset 1 found as a fallback when target >= 9
-            ([1], 16, 1),
-        ],
+            (
+                torch.fx.Node(
+                    graph=torch.fx.Graph(),
+                    name="aten::add.Tensor",
+                    op="call_function",
+                    target=torch.ops.aten.add.Tensor,  # type: ignore[attr-defined]
+                    args=(torch.tensor(3), torch.tensor(4)),
+                    kwargs={},
+                ),
+            ),
+            (
+                torch.fx.Node(
+                    graph=torch.fx.Graph(),
+                    name="aten::add.Tensor",
+                    op="call_function",
+                    target=torch.ops.aten.add.Tensor,  # type: ignore[attr-defined]
+                    args=(torch.tensor(3), torch.tensor(4)),
+                    kwargs={"alpha": 1},
+                ),
+            ),
+        ]
     )
-    def test_dispatch_opset_version_returns_correct_version(
-        self, available_opsets: Sequence[int], target: int, expected: int
+    def test_respect_custom_ops_in_find_the_perfect_or_nearest_match_onnxfunction(
```

Comment:
```suggestion
    def test_find_the_perfect_or_nearest_match_onnxfunction_gives_custom_ops_precedence(
```

consider: `method name_expected hebavior_<condition>`